tests/test_problem.py

```diff
@@ -84,3 +89,32 @@ def test_wrong_custom_sampling_logic(mode):
     }
     with pytest.raises(RuntimeError):
         poisson_problem.discretise_domain(sample_rules=sampling_rules)
+
+
+def test_aggregate_data():
+    poisson_problem = Poisson()
+    poisson_problem.conditions["data"] = Condition(
+        input=LabelTensor(torch.tensor([[0.0, 1.0]]), labels=["x", "y"]),
+        target=LabelTensor(torch.tensor([[0.0]]), labels=["u"]),
+    )
+    poisson_problem.discretise_domain(0, "random", domains="all")
+    poisson_problem.collect_data()
+    assert isinstance(poisson_problem.collected_data, dict)
+    for name, conditions in poisson_problem.conditions.items():
+        assert name in poisson_problem.collected_data.keys()
+        if isinstance(conditions, InputTargetCondition):
+            assert "input" in poisson_problem.collected_data[name].keys()
+            assert "target" in poisson_problem.collected_data[name].keys()
+        elif isinstance(conditions, DomainEquationCondition):
+            assert "input" in poisson_problem.collected_data[name].keys()
+            assert "target" not in poisson_problem.collected_data[name].keys()
+            assert "equation" in poisson_problem.collected_data[name].keys()
+
+
+def test_wrong_aggregate_data():
+    poisson_problem = Poisson()
+    poisson_problem.discretise_domain(0, "random", domains=["D"])
+    with pytest.raises(RuntimeError):
+        poisson_problem.collected_data()
```

Comment:
this will always raise error because collected_data is a property. I guess you want to check it is none.